server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -458,11 +511,42 @@ private void enable(boolean enabled) {
 
     private void updatePollInterval(TimeValue newInterval) {
         this.pollInterval = newInterval;
-        maybeScheduleJob();
+        // After the health logger is stopped we do not want to reschedule it
+        if (isStoppedOrClosed() == false) {
+            maybeScheduleJob();
+        }
+    }
+
+    private boolean isStarted() {
+        return lifecycleState() == Lifecycle.State.STARTED;
+    }
+
+    private boolean isStoppedOrClosed() {
+        return lifecycleState() == Lifecycle.State.STOPPED || lifecycleState() == Lifecycle.State.CLOSED;
+    }
+
+    // Visible for testing
+    TimeValue getPollInterval() {
+        return pollInterval;
+    }
+
+    // Visible for testing
+    boolean isHealthNode() {
+        return isHealthNode;
+    }
+
+    // Visible for testing
+    boolean enabled() {
+        return enabled;
+    }
+
+    // Visible for testing
+    boolean currentlyRunning() {
+        return currentlyRunning.availablePermits() == 0;
     }
```

Comment:
These methods are helper methods that expose either preconditions or the synchronisation state of the service to the test. I  want to use accessors instead of exposing the variable itself as it was, because I do not want anyone to be able to change their state. They are simply there so the test can verify things.